pkg/config/validation/validation.go

```diff
@@ -3477,3 +3479,53 @@ func validateTelemetryProviders(providers []*telemetry.ProviderRef) error {
 	}
 	return nil
 }
+
+// ValidateWasmPlugin validates a WasmPlugin.
+var ValidateWasmPlugin = registerValidateFunc("ValidateWasmPlugin",
+	func(cfg config.Config) (Warning, error) {
+		spec, ok := cfg.Spec.(*extensions.WasmPlugin)
+		if !ok {
+			return nil, fmt.Errorf("cannot cast to service entry")
+		}
+
+		errs := Validation{}
+		errs = appendValidation(errs,
+			validateWorkloadSelector(spec.Selector),
+			validateWasmPluginURL(spec.Url),
+			validateWasmPluginSHA(spec),
+		)
+		return errs.Unwrap()
+	})
+
+func validateWasmPluginURL(pluginURL string) error {
+	if pluginURL == "" {
+		return fmt.Errorf("url field needs to be set")
+	}
+	validSchemes := map[string]bool{
+		"": true, "file": true, "http": true, "https": true, "oci": true,
+	}
+
+	u, err := url.Parse(pluginURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse url: %s", err)
+	}
+	if _, found := validSchemes[u.Scheme]; !found {
+		return fmt.Errorf("url contains unsupported scheme: %s", u.Scheme)
+	}
+	return nil
+}
+
+func validateWasmPluginSHA(plugin *extensions.WasmPlugin) error {
+	if plugin.XSha256 == nil {
+		return nil
+	}
+	if len(plugin.GetSha256()) != 64 {
+		return fmt.Errorf("sha256 field must be 64 characters long")
+	}
+	for _, r := range plugin.GetSha256() {
+		if !('a' <= r && r <= 'z' || '0' <= r && r <= '9') {
+			return fmt.Errorf("sha256 field must match [a-z0-9]{64} pattern")
```

Comment:
sorry actually this should be in hex :D
```suggestion
		if !('a' <= r && r <= 'f') {
			return fmt.Errorf("sha256 field must match [a-f]{64} pattern")
```

Reply:
that's absolutely right :man_facepalming: